app/javascript/mastodon/features/follow_recommendations/index.js

```diff
@@ -40,6 +41,11 @@ class FollowRecommendations extends ImmutablePureComponent {
     if (suggestions.size === 0) {
       dispatch(fetchSuggestions(true));
     }
+
+    // Force the home timeline to be reloaded when the user navigates
+    // to it; if the user is new, it would've been empty before
+
+    dispatch(markAsPartial('home'));
```

Comment:
We should rather/also do it on unload or something, as doing this on load will not work.

Indeed, on app load, the Home feed is requested in all cases, causing the partial flag to get overridden by processing the response which will be received/processed after the component gets mounted.